cultivation/scripts/flashcore/database.py

```
@@ -0,0 +1,444 @@
+                new_review_id = int(result[0])
+                cursor.commit()
+            logger.info(f"Successfully added review with ID: {new_review_id} for card UUID: {review.card_uuid}")
+            return new_review_id
+        except duckdb.Error as e:
+            logger.error(f"Error adding review for card UUID {review.card_uuid}: {e}")
+            if 'cursor' in locals() and cursor.connection.is_open:
+                try:
+                    cursor.rollback()
+                except duckdb.Error as rb_err:
+                    logger.error(f"Error during rollback for add_review: {rb_err}")
+            raise ReviewOperationError(f"Failed to add review: {e}", original_exception=e)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)